app/src/behaviors/behavior_mouse_wheel.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2021 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_mouse_wheel
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+
+#include <zmk/event_manager.h>
+#include <zmk/events/keycode_state_changed.h>
+#include <zmk/behavior.h>
+#include <zmk/hid.h>
+#include <zmk/endpoints.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+static int behavior_mouse_wheel_init(const struct device *dev) { return 0; };
+
+static int on_keymap_binding_pressed(struct zmk_behavior_binding *binding,
+                                     struct zmk_behavior_binding_event event) {
+    LOG_DBG("position %d keycode 0x%02X", event.position, binding->param1);
+    int32_t x = (binding->param1 & 0xFF00) >> 8;
```

Comment:
Consider defining a macro to do the translation from param to x/y

Reply:
The tricky part here is I'm not sure how exactly the HID report handles speeds. `0xFFFF` is the smallest possible speed going up, and `0x0001` is the smallest speed for going down, so does the speed increase towards the middle? Of course, I may be stupid and this may be how the `uint32` to `int32` conversion works: I'll admit I haven't looked into this.

Reply:
0xFFFF = -1, have a look at "one's complement": https://www.electronics-tutorials.ws/binary/signed-binary-numbers.html

Reply:
I knew that, but the whole "converging towards the middle" part didn't click for some reason. I get it now, thanks!

Reply:
Implemented the macros.